module/os/linux/spl/spl-kmem.c

```diff
@@ -245,7 +245,14 @@ spl_kmem_alloc_impl(size_t size, int flags, int node)
 				return (NULL);
 			}
 		} else {
-			if (flags & KM_VMEM) {
+			/*
+			 * We only use kmalloc when doing
+			 * kmem_alloc(KM_NOSLEEP). Otherwise we use kvmalloc,
```

Comment:
I think it's worth mentioning that we don't do this for `kmem_alloc(KM_NOSLEEP)` because if we fallback to vmalloc, then we may sleep.